federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -158,6 +215,25 @@ func NewIngressController(client federation_release_1_4.Interface) *IngressContr
 			err := client.Extensions().Ingresses(ingress.Namespace).Delete(ingress.Name, &api.DeleteOptions{})
 			return err
 		})
+
+	// Federated configmap updater along with Create/Update/Delete operations.  Only Update should ever be called.
+	ic.federatedConfigMapUpdater = util.NewFederatedUpdater(ic.configMapFederatedInformer,
+		func(client kube_release_1_4.Interface, obj pkg_runtime.Object) error {
+			configMap := obj.(*api.ConfigMap)
+			configMapName := types.NamespacedName{Name: configMap.Name, Namespace: configMap.Namespace}
+			return fmt.Errorf("Internal error: Incorrectly attempting to create ConfigMap: %q", configMapName)
+		},
+		func(client kube_release_1_4.Interface, obj pkg_runtime.Object) error {
+			configMap := obj.(*v1.ConfigMap)
+			glog.V(4).Infof("Attempting to update ConfigMap: %v", configMap)
```

Comment:
changing the uid once you have working ingresses is not easy. It requires recreation of cloud resources and hance downtime. Typically users should not want to do this, the recommended way to handle it is to delete ingresses, change uid, recreate ingress (https://github.com/kubernetes/contrib/blob/master/ingress/controllers/gce/BETA_LIMITATIONS.md#changing-the-cluster-uid). It would be even worse if one cluster ended up with a different uid as it was creating ingress.

Can you complain loudly anytime we actually update it (event against master, Warningf logging)?